### PR TITLE
Add CF support to prop files

### PR DESCRIPTION
**Pull request by gregturn**
This patch will add Cloud Foundry property settings to the default configuration.

It also updates echo with slack settings.

### CODE REVIEW

File: config/clouddriver.yml [hunk 1]
```diff
@@ -40,3 +40,20 @@ google:
     - name: ${providers.google.primaryCredentials.name}
       project: ${providers.google.primaryCredentials.project}
       jsonPath: ${providers.google.primaryCredentials.jsonPath}
+
+cf:
+  enabled: ${providers.cf.enabled:false}
+
+  # TODO: Migrate this out of clouddriver and into either orca or igor
+  jenkins:
```
Comment (tomaslin) [inline config/clouddriver.yml]: Did you address @ewiseblatt comments on the other PR about having these settings here? Are they still needed ( jenkins )

File: config/clouddriver.yml [hunk 2]
```diff
@@ -40,3 +40,20 @@ google:
...
+    username: ${services.jenkins.defaultMaster.username}
+    password: ${services.jenkins.defaultMaster.password}
+
+  accounts:
+    - name: ${providers.cf.primaryCredentials.name}
+      api: ${providers.cf.primaryCredentials.api}
+      console: ${providers.cf.primaryCredentials.console}
+      org: ${providers.cf.defaultOrg}
+      space: ${providers.cf.defaultSpace}
+      username: ${cf.account.name}
```
Comment (duftler) [inline config/clouddriver.yml]: Where are these defined? I don't see `cf.account.name` or `cf.account.password`. Maybe I'm just looking in the wrong files.

Comment (tomaslin) [inline config/clouddriver.yml]: environment variables.

Comment (duftler) [inline config/clouddriver.yml]: So won't this fail if those env variables are not defined, and there are no defaults here?

File: config/clouddriver.yml (hunk 1 above)
Comment (gregturn) [inline config/clouddriver.yml]: I haven't fixed the core issue yet, but it's on my radar.

File: config/clouddriver.yml (hunk 2 above)
Comment (gregturn) [inline config/clouddriver.yml]: I should stick some NA defaults.

Comment (duftler) [inline config/clouddriver.yml]: Can't you just add an empty default like this one? https://github.com/spinnaker/spinnaker/blob/master/config/default-spinnaker-local.yml#L30

Comment (gregturn) [inline config/clouddriver.yml]: Ahh. Didn't know you could do that. Perfect.

Comment (ewiseblatt) [inline config/clouddriver.yml]: is it possible to make environment variables look like environment variables for clarity (UPPER_CASE)?

Comment (tomaslin) [inline config/clouddriver.yml]: wouldn't that actually stop you from adding them by hand in another yml file though?

File: config/default-spinnaker-local.yml [hunk 3]
```diff
@@ -103,5 +116,8 @@ services:
         account: # twilio account id
         token: # twilio auth token
         from: # phone number by which sms messages are sent
+      slack:
+        enabled: false
+        token: # the part after http://hooks.slack.com/services/
```
Comment (ewiseblatt) [inline config/default-spinnaker-local.yml]: Is this an independent change?
slack isnt tied to CF, is it?
If it is the other way around, the slack change can get in there first anyway.

File: config/echo.yml [hunk 4]
```diff
@@ -14,6 +14,10 @@ orca:
 
 endpoints.health.sensitive: false
 
+slack:
+  enabled: ${services.echo.slack.enabled:false}
```
Comment (ewiseblatt) [inline config/echo.yml]: service.notifications

File: config/default-spinnaker-local.yml (hunk 3 above)
Comment (tomaslin) [inline config/default-spinnaker-local.yml]: I would say slack should be a different PR

File: config/clouddriver.yml (hunk 1 above)
Comment (ewiseblatt) [inline config/clouddriver.yml]: Is this an independent change?
slack isnt tied to CF, is it?
If it is the other way around, the slack change can get in there first anyway.

File: config/clouddriver.yml (hunk 2 above)
Comment (ewiseblatt) [inline config/clouddriver.yml]: @tomaslin, you are right. I stand corrected.

Comment (gregturn) [inline config/clouddriver.yml]: FYI I'm not a fan of declaring UPPER_CASE env variables, when every single property you're seeing support overrides via the same method.

**Comment by gregturn** [comment]
I pulled slack stuff and will file a separate PR.

File: config/clouddriver.yml (hunk 2 above)
Comment (ewiseblatt) [inline config/clouddriver.yml]: I'm not a fan of environ variables at all but recognize they are common and
accepted practice.
I'm just noting the convention that is in use to keep things more
readable/predictable in the "global" namespace.
Especially since the possible sources of these values are sharded to the
wind.
Once you drop into a provider, you can (and should be able to) follow the
providers conventions or idiosyncrasies.

**Comment by gregturn** [comment]
See #610 for slack updates.

**Comment by anotherchrisberry** [comment]
What's the story on this PR? Is it still valid or should we close it?

**Comment by gregturn** [comment]
@anotherchrisberry I need to update and revise. The outstanding issues I mentioned earlier should now be resolved.

**Comment by gregturn** [comment]
Okay, I've updated and polished this PR. Please review.

**Comment by anotherchrisberry** [comment]
LGTM - @duftler @ewiseblatt @tomaslin any objections to merging?
